### PR TITLE
Run3-ft28 Correct in view of changing muon geometry

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Change in the builder of RPC/CSC geometries needs different set of parameters

#### PR validation:

Tested using runAll.csh in the test directory

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
-1
Sorry, no direct modification of the HLT menu dumps, instead, add a customisation function in
HLTrigger/Configuration/python/customizeHLTforCMSSW.py and call it in there as well.
Can the added parameter be taken care of by fillDescriptions of the plugins?

**Comment by bsunanda** [comment]
@Martin-Grunewald I am sending the customize function. How to update the scripts in the test area?

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -177,6 +177,32 @@ def customiseFor2017DtUnpacking(process):
 
     return process
 
+
+def customiseForMuonGeometry(process):
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:181]: Please use PR number in name of routine (see example)

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 2]
```diff
@@ -177,6 +177,32 @@ def customiseFor2017DtUnpacking(process):
...
+            useDD4hep = cms.bool( False ),
+            useCentreTIOffsets = cms.bool( False ),
+            applyAlignment = cms.bool( True )
+        )
+
+    return process
+
+
 # CMSSW version specific customizations
 def customizeHLTforCMSSW(process, menuType="GRun"):
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:208]: need to actually call the customisation routine here

**Comment by Martin-Grunewald** [comment]
Sorry, but this will not work in the long run with ConfDB parsing creating a new template to put these changes directly into the menus.

ConfDB parsing picks up the available parameters of a plugin from either fillDescriptions (does not exist for these two) or from an explicit cfi file in the python directory.

So far ConfBD parsing used (11_1_0_pre8):

RPCGeometryESModule                               Geometry/RPCGeometry/python/rpcGeometryInMTCC_cfi.py

CSCGeometryESModule                               Geometry/CSCGeometryBuilder/python/cscGeometry_cfi.py                                                                                            

These files would need update! The rps one also seems no longer to be there....????

**Comment by Martin-Grunewald** [comment]
-1

**Comment by bsunanda** [comment]
@Martin-Grunewald Now I understand and I have corrected the 2 cfi's. Have to do that for DT and GEM as well. I wonder which files are used for HCAL, ECAL and HGCAL. Could you send me the paths and I shall check the relevant correct paths. I am closing this PR which is indeed useless.

**Comment by Martin-Grunewald** [comment]
Hi @bsunanda 
grepping on GeometryESModule, these are the cfi files currently used.
This is flexible, you can remove them but there should be at least one to define the plugin so that it can be picked up.

grep GeometryESModule
MTDDigiGeometryESModule                           Geometry/MTDGeometryBuilder/mtdGeometryDB_cfi.py                                                                                                                                                        
MTDDigiGeometryESModule                           Geometry/MTDGeometryBuilder/mtdGeometry_cfi.py                                                                                                                                                          
TrackerDigiGeometryESModule                       Geometry/TrackerGeometryBuilder/trackerGeometryDB_cfi.py                                                                                                                                                
TrackerDigiGeometryESModule                       Geometry/TrackerGeometryBuilder/trackerGeometry_cfi.py                                                                                                                                                  
CSCGeometryESModule                               Geometry/CSCGeometryBuilder/python/cscGeometry_cfi.py                                                                                                                                                   
TrackerDigiGeometryESModule                       Geometry/TrackerGeometryBuilder/python/trackerGeometryDB_cfi.py                                                                                                                                         
DTGeometryESModule                                Geometry/DTGeometry/python/dtGeometry_cfi.py                                                                                                                                                            
GEMGeometryESModule                               Geometry/GEMGeometryBuilder/python/gemGeometryDB_cfi.py                                                                                                                                                 
ME0GeometryESModule                               Geometry/GEMGeometryBuilder/python/me0Geometry_cfi.py                                                                                                                                                   
RPCGeometryESModule                               Geometry/RPCGeometry/python/rpcGeometryInMTCC_cfi.py                                                                                                                                                    
CTPPSGeometryESModule                             Geometry/VeryForwardGeometry/python/geometryPPS_CMSxz_fromDD_2016_cfi.py

**Comment by bsunanda** [comment]
Thanks Martin

What about calorimeter geometry? Is it done using CaloEventSetup? Cheers

Sunanda
________________________________

**Comment by Martin-Grunewald** [comment]
CaloTowerConstituentsMapBuilder                   Geometry/CaloEventSetup/caloTowerConstituents_cfi.py                                                                                                                                                    
CaloGeometryBuilder                               Geometry/CaloEventSetup/python/CaloGeometryBuilder_cfi.py                                                                                                                                               
CastorGeometryToDBEP                              Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
CastorHardcodeGeometryEP                          Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
ZdcGeometryToDBEP                                 Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
EcalPreshowerGeometryToDBEP                       Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
HcalHardcodeGeometryEP                            Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
CaloTowerGeometryToDBEP                           Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
CaloTowerHardcodeGeometryEP                       Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
HcalGeometryToDBEP                                Geometry/CaloEventSetup/python/CaloGeometryDBWriter_cfi.py                                                                                                                                              
FastTimeTopologyBuilder                           Geometry/CaloEventSetup/python/FastTimeTopology_cfi.py                                                                                                                                                  
CaloTopologyBuilder                               Geometry/CaloEventSetup/python/CaloTopology_cfi.py                                                                                                                                                      
HGCalTopologyBuilder                              Geometry/CaloEventSetup/python/HGCalV9Topology_cfi.py                                                                                                                                                   
EcalTrigTowerConstituentsMapBuilder               Geometry/CaloEventSetup/python/EcalTrigTowerConstituents_cfi.py

**Pull request closed without merging**